### PR TITLE
Auto-fill schedule url in delegate report if exists

### LINKED ISSUE #3864: Delegate report schedule bar
**Is your feature request related to a problem? Please describe.**
The Delegate report allows for Delegates to paste a schedule link whereas the results submission forces use of the WCA schedule tool

**Describe the solution you'd like**
Have the schedule box auto-filled with the schedule of the competition

**Describe alternatives you've considered**
Nothing really alternative here :)

### PR DESCRIPTION
**Pull request by UnsolvedCypher**
Closes #3864

### CODE REVIEW

File: WcaOnRails/app/views/delegate_reports/edit.html.erb [hunk 1]
```diff
@@ -11,7 +11,8 @@
     <% if @delegate_report.posted? %>
       <%= f.input :discussion_url %>
     <% end %>
-    <%= f.input :schedule_url %>
+    <% default_schedule_url = @competition.competition_venues.any? ? link_to_competition_schedule_tab(@competition) : '' %>
+    <%= f.input :schedule_url, input_html: { value: default_schedule_url } %>
```
Comment (jonatanklosko) [inline WcaOnRails/app/views/delegate_reports/edit.html.erb]: Given this forces WCA schedule URL, perhaps we should disable that field when using `default_schedule_url`?

File: WcaOnRails/app/views/delegate_reports/edit.html.erb [hunk 2]
```diff
@@ -11,7 +11,8 @@
     <% if @delegate_report.posted? %>
       <%= f.input :discussion_url %>
     <% end %>
-    <%= f.input :schedule_url %>
+    <% default_schedule_url = @competition.competition_venues.any? ? link_to_competition_schedule_tab(@competition) : '' %>
```
Comment (jfly) [inline WcaOnRails/app/views/delegate_reports/edit.html.erb]: Can you introduce a `has_schedule?` helper method to our competitions model? I don't think this code should be aware of the that that we store schedules as competition_venues.

Comment (viroulep) [reply WcaOnRails/app/views/delegate_reports/edit.html.erb]: It [exists](https://github.com/thewca/worldcubeassociation.org/blob/a19005dce0cacd1225273f8b443b36fd2c689651/WcaOnRails/app/models/competition.rb#L710) ;)

File: WcaOnRails/app/views/delegate_reports/edit.html.erb (hunk 1 above)
Comment (UnsolvedCypher) [reply WcaOnRails/app/views/delegate_reports/edit.html.erb]: I'm a bit confused- are you saying that if a WCA schedule URL exists already we don't need to have this field and we should just automatically link to it?

File: WcaOnRails/app/views/delegate_reports/edit.html.erb (hunk 2 above)
Comment (UnsolvedCypher) [reply WcaOnRails/app/views/delegate_reports/edit.html.erb]: Thank you, that is much cleaner. I am now using that helper method :)

File: WcaOnRails/app/views/delegate_reports/edit.html.erb (hunk 1 above)
Comment (jfly) [reply WcaOnRails/app/views/delegate_reports/edit.html.erb]: That feels reasonable to me. Are there situations where we'd want to let people enter something other than the WCA schedule here, if a WCA schedule exists?

Comment (jonatanklosko) [reply WcaOnRails/app/views/delegate_reports/edit.html.erb]: Ah sorry I missed your comment! My point was just that if there is WCA schedule and we fill this field, it's certainly the URL we want (so it makes sense to disable it).

Comment (UnsolvedCypher) [reply WcaOnRails/app/views/delegate_reports/edit.html.erb]: By "disable" do you mean use the disabled HTML attribute, or just omit the field entirely?

Comment (jonatanklosko) [reply WcaOnRails/app/views/delegate_reports/edit.html.erb]: I mean just disabling the element, although I imagine us getting rid of it in the future ^^

Comment (jonatanklosko) [reply WcaOnRails/app/views/delegate_reports/edit.html.erb]: I mean just disabling the element, although I imagine us getting rid of it in the future ^^

**Comment by AlbertoPdRF** [comment]
Is there something left to do here or can the PR be merged?

File: WcaOnRails/app/views/delegate_reports/edit.html.erb (hunk 1 above)
Comment (viroulep) [reply WcaOnRails/app/views/delegate_reports/edit.html.erb]:
```suggestion
    <%= f.input :schedule_url, input_html: { value: default_schedule_url }, disabled: default_schedule_url.present? %>
```

I think this should work out fine :)

Comment (AlbertoPdRF) [reply WcaOnRails/app/views/delegate_reports/edit.html.erb]: I can't commit to @UnsolvedCypher's repository, so to amend this we either need Matt to fix it or a brand new PR. I can take care of the latter if necessary :)

**Comment by AlbertoPdRF** [comment]
Superseded by #4279.

**Pull request closed without merging**